engine/src/flutter/impeller/renderer/backend/vulkan/descriptor_pool_vk.cc

```diff
@@ -111,6 +86,10 @@ fml::StatusOr<vk::DescriptorSet> DescriptorPoolVK::AllocateDescriptorSets(
     set_info.setDescriptorPool(pools_.back().get());
     result = context_vk.GetDevice().allocateDescriptorSets(&set_info, &set);
   }
+  if (existing == descriptor_sets_.end()) {
+    descriptor_sets_[pipeline_key] = DescriptorCache{};
+  }
```

Comment:
use `try_emplace` to avoid double lookups

Reply:
I need some pointers here...

Reply:
```c++
descriptor_sets_.try_emplace(pipeline_key, DescriptorCache{});
result.first->used.push_back(set);
```

Reply:
actually `lookup_result.first->second.used.push_back(set);`

Its an iterator to an iterator. What a language.